java/pubsub/src/main/java/com/example/pubsub/OpenCensusStatsUtil.java

```diff
@@ -0,0 +1,135 @@
+  private static final TagKey METHOD = TagKey.create("METHOD");
+
+  // The method key allows us to break down the recorded latencies.
+  private static final TagKey PUBLISHER = TagKey.create("PUBLISHER");
+
+  private static final TagKey SUBSCRIBER = TagKey.create("SUBSCRIBER");
+
+  private static int pubCount = 0;
+
+  public static Scope createPublisherScope() {
+    TagValue val = TagValue.create("publisher-" + pubCount);
+    pubCount = (pubCount + 1) % 10;
+    return tagger.currentBuilder().put(PUBLISHER, val).buildScoped();
+  }
+
+  private static int subCount = 0;
+
+  public static Scope createSubscriberScope() {
+    TagValue val = TagValue.create("subscriber-" + subCount);
+    subCount = (subCount + 1) % 10;
+    return tagger.currentBuilder().put(SUBSCRIBER, val).buildScoped();
+  }
+
+  /**
+   * Creates a Scope that captures the current system time and records latency when Scope is
+   * close()'d.
+   */
+  public static final Scope createLatencyScope() {
+    return new LatencyStatsRecorder();
+  }
+
+  /*
+   * A utility class to simplify recording latency. When a LatencyStatsRecorder is created,
+   * the current time is captured and the specified key/value pair are added to the current
+   * tag scope. When the close method is called, the latency is calculated and recorded
+   * against the LATENCY measure (and, implicitly, the specified tag).
+   */
+  private static final class LatencyStatsRecorder implements Scope {
+    private final long startTimeMs;
+
+    private LatencyStatsRecorder() {
+      this.startTimeMs = System.currentTimeMillis();
+    }
+
+    /**
+     * On close, records latency against the LATENCY measure.
+     */
+    @Override
+    public void close() {
+      long latencyMs = System.currentTimeMillis() - startTimeMs;
+      statsRecorder.newMeasureMap().put(LATENCY, latencyMs).record();
+    }
+  }
+
+  // Measure for Pub/Sub latency in milliseconds.
+  private static final Measure.MeasureDouble LATENCY =
+      Measure.MeasureDouble.create("my.io/cloud-pubsub/latency", "Latency in milliseconds.", "ms");
+
+  private static final View LATENCY_VIEW =
+      View.create(
+          View.Name.create("my.io/cloud-pubsub/latency"),
+          "Latency in milliseconds",
+          LATENCY,
+          Distribution.create(BucketBoundaries.create(Arrays.asList(
+              1.0, 2.0, 4.0, 8.0, 16.0, 32.0, 64.0, 128.0, 256.0, 512.0, 1024.0, 2048.0, 4096.0))),
+          Arrays.asList(PUBLISHER, SUBSCRIBER));
+
+  static {
+    if (!PROJECT_ID.isEmpty()) {
+      // Register latency view.
+      Stats.getViewManager().registerView(LATENCY_VIEW);
+      try {
+        // Initialize stats exporter.
+        StackdriverStatsExporter.createAndRegister(
+            StackdriverStatsConfiguration.builder()
+            .setProjectId(PROJECT_ID)
+            .setExportInterval(Duration.create(15, 0))
+            .build());
+      } catch (IOException exn) {
+        if (exn == null) {
```

Comment:
Will `exn` be null?

Reply:
No - was using this to prevent a compiler warning - removed.